### PR TITLE
Case cost model - frontend

### PR DESCRIPTION
**Pull request by metroid-samus**
This PR adds the ability to create case costs by assigning a cost model to case types. All cases of the same type use the same cost model to calculate ongoing response costs.

The final cost of a case is an accumulation of one or multiple cost types.
Screenshot 2024-08-28 at 8 07 30 PM 
Screenshot 2024-08-28 at 8 17 28 PM

Each case type can be assigned a cost model, which are used to calculate ongoing response costs. If no cost model is assigned to a case type, all cases of that same type do not update their response costs.
Screenshot 2024-08-28 at 8 10 44 PM

### CODE REVIEW

File: src/dispatch/case_cost/views.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+from .service import create, delete, get, update
+
+
+router = APIRouter()
+
+
+@router.get("", response_model=CaseCostPagination)
+def get_case_costs(common: CommonParameters):
+    """Get all case costs, or only those matching a given search term."""
+    print("get case costs")
```
Comment (wssheldon) [inline src/dispatch/case_cost/views.py:23]:
```suggestion
```

**Review by wssheldon — APPROVED** [review]

File: src/dispatch/case_cost/views.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+
+@router.get("/{case_cost_id}", response_model=CaseCostRead)
+def get_case_cost(db_session: DbSession, case_cost_id: PrimaryKey):
+    """Get an case cost by its id."""
+    case_cost = get(db_session=db_session, case_cost_id=case_cost_id)
+    if not case_cost:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=[{"msg": "An case cost with this id does not exist."}],
```
Comment (mvilanova) [inline src/dispatch/case_cost/views.py:34]:
```suggestion
            detail=[{"msg": "A case cost with this id does not exist."}],
```

Comment (mvilanova) [reply src/dispatch/case_cost/views.py:34]: There are other docstrings with the same issue.

File: src/dispatch/case_cost_type/views.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+@router.get("", response_model=CaseCostTypePagination)
+def get_case_cost_types(common: CommonParameters):
+    """Get all case cost types, or only those matching a given search term."""
+    return search_filter_sort_paginate(model="CaseCostType", **common)
+
+
+@router.get("/{case_cost_type_id}", response_model=CaseCostTypeRead)
+def get_case_cost_type(db_session: DbSession, case_cost_type_id: PrimaryKey):
+    """Get an case cost type by its id."""
```
Comment (mvilanova) [inline src/dispatch/case_cost_type/views.py:28]:
```suggestion
    """Get a case cost type by its id."""
```

Comment (mvilanova) [reply src/dispatch/case_cost_type/views.py:28]: May want to do a search and replace to fix all of them at once.

Comment (metroid-samus) [reply src/dispatch/case_cost_type/views.py]: Good catch. Done.

**Pull request merged**